tests/test_dynamodb.py

```diff
@@ -528,3 +587,29 @@ def test_alias_model_validator_ingest(dynamo):
     data.pop("typ")
     with pytest.raises(ValidationError):
         AliasKeyModel(**data)
+
+
+def test_batch_write(dynamo, complex_table):
+    response = {"UnprocessedItems": {}}
+    data = [ComplexKeyModel.parse_obj(complex_model_data_generator()) for x in range(0, 10)]
+    un_proc = ComplexKeyModel.batch_save(data)
+    assert un_proc == response["UnprocessedItems"]
+    res_get = ComplexKeyModel.get((data[0].account, data[0].sort_date_key))
+    res_query = ComplexKeyModel.query(
+        Rule(f"account == '{data[0].account}' and sort_date_key == '{data[0].sort_date_key}'")
+    )
+    assert res_get == data[0]
+    assert res_query.count == 1
+    assert res_query.records == [data[0]]
+
+
+def test_message_batch_write_client_exception(dynamo, complex_table):
+    data = [
+        ComplexKeyModel.parse_obj(complex_model_data_generator(body="some big string" * 10000))
+        for x in range(0, 2)
+    ]
+    with pytest.raises(ClientError) as exc:
+        ComplexKeyModel.batch_save(data)
+    assert (
+        exc.value.response["Error"]["Message"] == "Item size has exceeded the maximum allowed size"
```

Comment:
Ideally, we should normalize this error...but it's not a blocker for today.

Reply:
agree.